ompi/debuggers/ompi_msgq_dll.c

```diff
@@ -682,8 +684,21 @@ static int rebuild_communicator_list (mqs_process *proc)
                                     p_info );
             old->group = find_or_create_group( proc, group_base );
         }
-        mqs_fetch_data( proc, comm_ptr + i_info->ompi_communicator_t.offset.c_name,
-                        64, old->comm_info.name );
+        mqs_taddr_t name_addr = ompi_fetch_pointer( proc,
+                                                    comm_ptr + i_info->ompi_communicator_t.offset.c_name,
+                                                    p_info );
+        /* c_name can be up to MPI_MAX_OBJECT_NAME bytes, but we only
+         * copy the first (sizeof(old->comm_info.name)-1) here.  Make
+         * sure the string is correctly terminated. */
+        mqs_fetch_data( proc, name_addr, 64, old->comm_info.name );
+        size_t target_size = sizeof(old->comm_info.name);
+        old->comm_info.name[target_size - 1] = '\0';
+        /* Defensively zero anything beyond the actual name */
+        size_t src_strlen = strlen(old->comm_info.name);
+        if (target_size - 1 > src_strlen) {
```

Comment:
This condition can never happen, the line above sets the termination at position `target_size - 1` so the strlen cannot return anything larger than that. When they are equal memset will not do anything, so this test is unnecessary.